internal/core/components/rewriter.go

```diff
@@ -0,0 +1,91 @@
+package components
+
+import (
+	"context"
+	"runtime"
+	"sync"
+
+	"github.com/api7/apisix-seed/internal/core/entity"
+	"github.com/api7/apisix-seed/internal/core/storer"
+	"github.com/api7/apisix-seed/internal/discoverer"
+	"github.com/api7/apisix-seed/internal/utils"
+)
+
+type Rewriter struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	// Limit the number of simultaneously update
+	sem chan struct{}
+}
+
+func (r *Rewriter) Init() {
+	r.ctx, r.cancel = context.WithCancel(context.TODO())
+	r.sem = make(chan struct{}, runtime.GOMAXPROCS(0)+10)
+
+	// Watch for service updates from Discoverer
+	for _, dis := range discoverer.GetDiscoverers() {
+		ch := dis.Watch()
+		go func() {
+			for {
+				select {
+				case <-r.ctx.Done():
+					return
+				case watch := <-ch:
+					values, entities, nodes, err := watch.Decode()
+					if err != nil {
+						continue
+					}
+
+					if values[0] == utils.EventUpdate {
```

Comment:
Look like we should also rewrite the newly added entity?

Reply:
The way it is handled now is that all messages sent by the Discoverer to the Rewriter are update messages, as https://github.com/api7/apisix-seed/blob/main/internal/discoverer/discoverer.go#L62 . I considered distinguishing the different cases in Discoverer later on.